google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -153,6 +159,14 @@ public Builder setSessionPoolOption(SessionPoolOptions sessionPoolOptions) {
     }
 
     /**
+     * Sets the labels to add to all Sessions created in this client. 
+     */
+    public Builder setSessionLabels(Map<String, String> sessionLabels) {
```

Comment:
Do we allow null values (and coerce them to empty) or require that the user give us empty string values in the first place?

Also, if we are given an invalid label, how is that revealed to the user?

Reply:
I was thinking that we will not do anything fancy and just pass the map as it is. The CreateSession call would fail if there is something illegal here and users would get a SpannerException whenever they do any operation on DatabaseClient (just like they do in case of errors like permission denied).

Reply:
Added null check and linked to the sessions doc for valid syntax.